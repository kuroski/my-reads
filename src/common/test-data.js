const testBooks = {
  books: [
    {
      title: 'Book Title 1',
      subtitle: 'Book Subtitle 1',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      authors: ['Author One', 'Author Two'],
      imageLinks: { thumbnail: 'http://img01.jpg' },
      id: 'PGR2AwAAQBAJ',
      shelf: 'wantToRead'
    },
    {
      title: 'Book Title 2',
      subtitle: 'Book Subtitle 2',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit2.',
      authors: ['Mist Misterious'],
      imageLinks: { thumbnail: 'http://img02.jpg' },
      id: 'PGR2AwAAQBAJ2',
      shelf: 'read'
    },
    {
      title: 'Book Title 3',
      subtitle: 'Book Subtitle 3',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit3.',
      authors: ['Lucky Guy', 'Telonious Monk', 'Rapid Fast'],
      imageLinks: { thumbnail: 'http://img03.jpg' },
      id: 'PGR2AwAAQBAJ3',
      shelf: 'wantToRead'
    }
  ]
};

export { testBooks };
