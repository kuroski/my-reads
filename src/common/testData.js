import { shelfState } from './shelfState';

const testBooks = {
  books: [
    {
      title: 'Book Title 1',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      authors: ['Author One', 'Author Two'],
      imageLinks: { thumbnail: 'http://img01.jpg' },
      id: 'PGR2AwAAQBAJ',
      shelf: shelfState.WANT_TO_READ
    },
    {
      title: 'Book Title 2',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit2.',
      authors: ['Mist Misterious'],
      imageLinks: { thumbnail: 'http://img02.jpg' },
      id: 'PGR2AwAAQBAJ2',
      shelf: shelfState.READ
    },
    {
      title: 'Book Title 3',
      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit3.',
      authors: ['Lucky Guy', 'Telonious Monk', 'Rapid Fast'],
      imageLinks: { thumbnail: 'http://img03.jpg' },
      id: 'PGR2AwAAQBAJ3',
      shelf: shelfState.WANT_TO_READ
    }
  ]
};

export { testBooks };
