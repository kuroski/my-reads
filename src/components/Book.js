import React from 'react';
import PropTypes from 'prop-types';

const Book = props => (
  <div className="book">
    <div className="book-top">
      <img
        src={props.coverImage}
        width="128"
        height="193"
        className="book-cover"
      />
      <div className="book-shelf-changer">
        <select>
          <option value="none" disabled>
            Move to...
          </option>
          <option value="currentlyReading">Currently Reading</option>
          <option value="wantToRead">Want to Read</option>
          <option value="read">Read</option>
          <option value="none">None</option>
        </select>
      </div>
    </div>
    <div className="book-title">To Kill a Mockingbird</div>
    <div className="book-authors">Harper Lee</div>
  </div>
);

Book.propTypes = {
  coverImage: PropTypes.string
};

Book.defaultProps = {
  coverImage: 'http://i.imgur.com/J5LVHEL.jpg'
};

export default Book;
