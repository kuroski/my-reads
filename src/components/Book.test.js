import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import Book from './Book';
import { testBooks } from '../common/test-data';

describe('Book Component', () => {
  let book;
  let wrapper;
  let mounted;

  beforeAll(() => {
    book = testBooks.books[0];

    const BookComponent = <Book coverImage={book.imageLinks.thumbnail} />;
    wrapper = shallow(BookComponent);
    mounted = mount(BookComponent);
  });

  it('renders correctly', () => {
    const BookTree = renderer.create(<Book />).toJSON();
    expect(BookTree).toMatchSnapshot();
  });

  it('renders the book cover', () => {
    expect(
      wrapper.containsMatchingElement(<img src={book.imageLinks.thumbnail} />)
    ).toBe(true);
  });
});
