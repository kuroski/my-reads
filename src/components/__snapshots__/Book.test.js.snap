// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Component renders correctly 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <img
      className="book-cover"
      height="193"
      src="http://i.imgur.com/J5LVHEL.jpg"
      width="128"
    />
    <div
      className="book-shelf-changer"
    >
      <select>
        <option
          disabled={true}
          value="none"
        >
          Move to...
        </option>
        <option
          value="currentlyReading"
        >
          Currently Reading
        </option>
        <option
          value="wantToRead"
        >
          Want to Read
        </option>
        <option
          value="read"
        >
          Read
        </option>
        <option
          value="none"
        >
          None
        </option>
      </select>
    </div>
  </div>
  <div
    className="book-title"
  >
    To Kill a Mockingbird
  </div>
  <div
    className="book-authors"
  >
    Harper Lee
  </div>
</div>
`;
